### PR TITLE
upast: 0.1 create/delete User impl (Dto/Service/Controller)

### PR DESCRIPTION
**Pull request by upastushyna**
add DTO and Service and Controller on backend with add/delete user

### CODE REVIEW

File: mentionserver/src/main/java/com.mention/dto/UserDtoNewUserRq.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import lombok.Data;
+
+@Data
+public class UserDtoNewUserRq {
+
+  private String username;
+  private String email;
+  private String password;
+  // we need parameter 'isActive' in this Dto ?
```
Comment (kKhairulin) [inline mentionserver/src/main/java/com.mention/dto/UserDtoNewUserRq.java]: yes, we need

File: mentionserver/src/main/java/com.mention/service/UserNewUser.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.mention.service;
+
+import com.mention.dto.UserDtoIdRq;
+import com.mention.dto.UserDtoNewUserRq;
+
+public interface UserNewUser {
```
Comment (kKhairulin) [inline mentionserver/src/main/java/com.mention/service/UserNewUser.java]: It's a service. name it like we name all services. (UserDtoRq, UserService, UserController) or (UserDataDtoRq, UserDataService, UserDataController). Remove old UserController if you chose the first naming option. Your controller and service will update, create and delete, so NewUser is a bad name for this functionality

File: mentionserver/src/main/java/com.mention/service/UserNewUserImpl.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  public UserNewUserImpl(UserRepository userRepository) {
+    this.userRepository = userRepository;
+  }
+
+  @Override
+  @Transactional
+  public void createNewUser(UserDtoNewUserRq userDtoNewUser) {
+    ModelMapper modelMapper = new ModelMapper();
+    User insertUser = modelMapper.map(userDtoNewUser, User.class);
+    insertUser.setActive(true);
```
Comment (kKhairulin) [inline mentionserver/src/main/java/com.mention/service/UserNewUserImpl.java]: remove this. we will set this parameter on front-end

File: mentionserver/src/main/java/com.mention/service/UserNewUserImpl.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    userRepository.save(insertUser);
+  }
+
+  @Override
+  @Transactional
+  public void deleteUser(UserDtoIdRq user) {
+    ModelMapper modelMapper = new ModelMapper();
+    Optional<User> currentUser = userRepository.findById(user.getId());
+    if (currentUser.isPresent() && currentUser.get().isActive()) {
```
Comment (kKhairulin) [inline mentionserver/src/main/java/com.mention/service/UserNewUserImpl.java]: We don't need to check if user is active here

File: mentionserver/src/main/java/com.mention/controller/UserCreateUserController.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import com.mention.dto.UserDtoNewUserRq;
+import com.mention.service.UserNewUserImpl;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/registration")
```
Comment (kKhairulin) [inline mentionserver/src/main/java/com.mention/controller/UserCreateUserController.java]: just name it /api/users

File: mentionserver/src/main/java/com.mention/service/UserNewUserImpl.java (hunk 3 above)
Comment (upastushyna) [reply mentionserver/src/main/java/com.mention/service/UserNewUserImpl.java]: what exactly will be set on front ? you mean this line ?
insertUser.setActive(true);

File: mentionserver/src/main/java/com.mention/service/UserNewUserImpl.java (hunk 4 above)
Comment (upastushyna) [reply mentionserver/src/main/java/com.mention/service/UserNewUserImpl.java]: ok. where we must check it?

File: mentionserver/src/main/java/com.mention/service/UserNewUserImpl.java (hunk 3 above)
Comment (kKhairulin) [reply mentionserver/src/main/java/com.mention/service/UserNewUserImpl.java]: Its void. Nothing will be sent on front-end. Yes, remove this line

File: mentionserver/src/main/java/com.mention/controller/UserCreateUserController.java (hunk 5 above)
Comment (upastushyna) [reply mentionserver/src/main/java/com.mention/controller/UserCreateUserController.java]: users or user ?

Comment (upastushyna) [reply mentionserver/src/main/java/com.mention/controller/UserCreateUserController.java]: in previus ver. UserController we used api/user

File: mentionserver/src/main/java/com.mention/service/UserNewUser.java (hunk 2 above)
Comment (upastushyna) [reply mentionserver/src/main/java/com.mention/service/UserNewUser.java]: absolutely agree with u, my friend

**Pull request merged**